src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,52 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents an Address's block number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBlock(String)}
+ */
+public class Block {
+    public static final String EXAMPLE = "123";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS = "Address block should only contain only numbers";
+    public static final String BLOCK_VALIDATION_REGEX = "\\d+";
```

Comment:
Good job adding validation to each component! (some block numbers have letters, but this is fine)